testcase.py

```diff
@@ -288,12 +291,15 @@ def advance(self, u):
                 self.tic_time = time.time()
                 # Get full current state
                 payload = self._get_full_current_state()
-                message = "Advanced simulation successfully."
+                if alert_message == '':
+                    message = "Advancing simulation successfully."
+                else:
+                    message = alert_message               
                 logging.info(message)
                 return status, message, payload
 
             else:
-                # Error in
+                # Errors in the simulation
                 status = 500
                 message = "Failed to advance simulation: {}"
```

Comment:
@SenHuang19 I think this needs a .format(res) at the end so the full error message is returned, just like it is logged in Line 306?